CMakeLists.txt

```diff
@@ -215,9 +227,8 @@ else(USE_SYSTEM_YAML_CPP)
         #get_property(yaml-cpp-pm_INCLUDE TARGET yaml-cpp-pm PROPERTY INCLUDE_DIRECTORIES)
         #include_directories(${yaml-cpp-pm_INCLUDE})
 
-        get_property(yaml-cpp-pm_LIB TARGET yaml-cpp-pm PROPERTY LOCATION)
-        set (EXTERNAL_LIBS ${EXTERNAL_LIBS} ${yaml-cpp-pm_LIB} )
-        set (EXTRA_DEPS ${EXTRA_DEPS} yaml-cpp-pm)
+        list(APPEND EXTERNAL_LIBS $<TARGET_FILE:yaml-cpp-pm>)
```

Comment:
This generator only works during build time. Does that fulfill the purpose of the expression?

If I'm not mistaken, before we were fetching the location of yaml_cpp. Is it possible to replace that with a standard cmake macro as find_package?

Reply:
yes, this is another obscure thing that would require much more work to make better: PM is using an embedded version of the yamp-cpp library, that's the reason of this expression. 
Look at "USE_SYSTEM_YAML_CPP"...

There are *many* things that could be done in more modern-cmake ways, but I don't have the cycles to work on that now, and it has the risk of breaking existing stuff :-)